### PR TITLE
Import Taproot pubkey

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/6657.

~Depends on https://github.com/btcsuite/btcwallet/pull/806.~

## Change Description
Allows public keys to be imported as BIP-0086 Taproot public keys through `WalletKit.ImportPublicKey`.
Any other use case (non BIP-0086) for Taproot scripts can use the new `WalletKit.ImportTapscript` RPC that allows Taproot outputs to be constructed from different levels of knowledge about the internal key and the script tree and then track those outputs on chain.

### CODE REVIEW

File: lnrpc/walletrpc/walletkit.proto [hunk 1]
```diff
@@ -100,7 +100,11 @@ service WalletKit {
     rpc ImportAccount (ImportAccountRequest) returns (ImportAccountResponse);
 
     /*
-    ImportPublicKey imports a public key as watch-only into the wallet.
+    ImportPublicKey imports a public key as watch-only into the wallet. The
+    public key is converted into a simple address of the given type and that
+    address script is watched on chain. For Taproot keys, this will only watch
+    the BIP-0086 style output script. Use ImportTapscript for more advanced key
```
Comment (Roasbeef) [inline lnrpc/walletrpc/walletkit.proto:106]: 👍

File: lntest/itest/lnd_taproot_test.go [hunk 2]
```diff
@@ -55,6 +56,11 @@ func testTaproot(net *lntest.NetworkHarness, t *harnessTest) {
 	testTaprootMuSig2KeySpendRootHash(ctxt, t, net.Alice, net)
 	testTaprootMuSig2ScriptSpend(ctxt, t, net.Alice, net)
 	testTaprootMuSig2CombinedLeafKeySpend(ctxt, t, net.Alice, net)
+
+	testTaprootImportTapscriptFullTree(ctxt, t, net.Alice, net)
```
Comment (Roasbeef) [inline lntest/itest/lnd_taproot_test.go:60]: Great additional test coverage here 😎

**Review by Roasbeef — COMMENTED** [review]
Straight forward diff! I dig the additional e2e tests as well, which 'll give us more confidence when integrating into that new project

**Comment by guggero** [comment]
Updated to merged `btcwallet` dependency. This is ready for full review.

**Review by sputn1ck — APPROVED** [review]
Very nice PR and clear commit structure!

I was wondering if it would it be possible to add information/conditions on how to spend tapscripts to the lnwallet? An example usecase would be the Loop-client not needing to sweep a transaction, just pass the spending conditions to LND. Might help devs build cool applications without having to watch the chain

**Comment by guggero** [comment]
I noticed we were missing the type `lnrpc.OutputScriptType_SCRIPT_TYPE_WITNESS_V1_TAPROOT` for wallet transaction outputs of the `GetTransactions` RPC. Added a commit and itest case for it.

**Review by Roasbeef — APPROVED** [review]
LGTM 🫕

**Comment by Roasbeef** [comment]
Needs a rebase, then this is g2g.

**Pull request merged**